### PR TITLE
fix(emotion): Cache changes

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Adds prefixer back to our stylis plugins (thanks emotion...).
* Updates overrides to also add our own plugins (before our default transforms).

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.34-alpha.ed50f3.0<br/>@codecademy/gamut-labs@9.3.4-alpha.ed50f3.0<br/>@codecademy/gamut-styles@7.7.4-alpha.ed50f3.0<br/>@codecademy/gamut-tests@1.1.3-alpha.ed50f3.0<br/>@codecademy/gamut@21.3.7-alpha.ed50f3.0<br/>@codecademy/markdown-overrides@0.4.8-alpha.ed50f3.0<br/>@codecademy/styleguide@25.2.4-alpha.ed50f3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6025885ecc2f690ab181afe0--gamut-preview.netlify.app>https://6025885ecc2f690ab181afe0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6025885ecc2f690ab181afe0>Deploy Logs</a></p>

File: packages/gamut-styles/src/cache/index.ts [hunk 1]
```diff
@@ -21,11 +22,15 @@ const getEmotionNode = () => {
...
     key: EMOTION_KEY,
     speedy: true,
     container: getEmotionNode(),
-    stylisPlugins: [focusVisible],
-    ...optionOverrides,
+    ...overrides,
+    stylisPlugins: [
+      ...(overrides?.stylisPlugins ?? []),
+      focusVisible,
+      prefixer as StylisPlugin,
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/cache/index.ts:34]: One wonders: if it wasn't an issue before...

**Review by JoshuaKGoldberg — APPROVED** [review]
...maybe we don't need this? 😄

**Review by jakemhiller — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.4.47-alpha.e73e7a.0<br/>@codecademy/gamut-labs@10.1.1-alpha.e73e7a.0<br/>@codecademy/gamut-styles@8.1.1-alpha.e73e7a.0<br/>@codecademy/gamut-tests@2.0.3-alpha.e73e7a.0<br/>@codecademy/gamut@25.0.1-alpha.e73e7a.0<br/>@codecademy/markdown-overrides@0.5.3-alpha.e73e7a.0<br/>@codecademy/styleguide@29.1.1-alpha.e73e7a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60300579cb9bec0157cba62d--gamut-preview.netlify.app>https://60300579cb9bec0157cba62d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60300579cb9bec0157cba62d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6035680e6808cb0097a65b13--gamut-preview.netlify.app>https://6035680e6808cb0097a65b13--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6035680e6808cb0097a65b13>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.7.1-alpha.1def41.0<br/>@codecademy/gamut-labs@10.1.9-alpha.1def41.0<br/>@codecademy/gamut-styles@8.1.1-alpha.1def41.0<br/>@codecademy/gamut-tests@2.1.1-alpha.1def41.0<br/>@codecademy/gamut@25.3.2-alpha.1def41.0<br/>@codecademy/markdown-overrides@0.5.3-alpha.1def41.0<br/>@codecademy/styleguide@29.4.2-alpha.1def41.0<br/></pre>

**Pull request merged**